Sources/RoutingKit/TrieRouter.swift

```diff
@@ -41,8 +41,14 @@ public final class TrieRouter<Output>: Router, CustomStringConvertible {
 
         // for each dynamic path in the route get the appropriate
         // child generating a new one if necessary
-        for component in path {
-            current = current.buildOrFetchChild(for: component, options: self.options)
+        for (index, component) in path.enumerated() {
+            switch component {
+            case .catchall:
+                assert(index == path.count - 1, "Catchall (aka. \(component)) can only be used as the ending")
```

Comment:
I'd prefer `precondition` here since this should also be a failure in release mode.

Reply:
Suggestion for failure message:
```
Catchall ('**') must be the last component in a path. 
```

Reply:
Got it. I’ll recommit and force-push in 12 hours since it’s late at night here where I am.